routing/router.go

```diff
@@ -1818,6 +1820,12 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 				update *lnwire.ChannelUpdate,
 				pubKey *btcec.PublicKey) {
 
+				err := ValidateChannelUpdateAnn(pubKey, update)
```

Comment:
Why is this added here? Validation is also done inside `r.applyChannelUpdate`.

Reply:
I agree this is ugly. My excuse it that I'm trying to minimize the changes I do in code to those mandatory for this PR. Therefore, if validation fails, processChannelUpdateAndRetry should immediately return, and should not update paySession at all (neither update hints or report edge policy failure), so the check must be there. However we can not rely on the error returned by applyChannelUpdate, as it may well error in other scenarios (i.e. not validation errors) which are perfectly ok, and do require retries. Initially I thought of doing the validation much earlier, i.e. before the switch on the error types, but this looks like a change that is too major for this PR; hence my feeling was that adding validation here is the right way to go, with only bad consequence being that its checked twice.

Reply:
I think it is ok to not update the payment session if the graph update fails. `ReportEdgeFailure` is important to call in case of a validation error. If we get an incorrect update we don't want to try that channel anymore.

Reply:
Ok. I removed the extra validation and returned if applyChannelUpdate fails after the edge is pruned.

I'm uncomfortable with the handling of error messages that fail in signature validation, as they do prune edges. Doesn't that allow an attacker to send error messages and cause the routing to fail?